crates/core/tedge_agent/src/main.rs

```diff
@@ -44,14 +44,25 @@ pub struct AgentOpt {
 #[tokio::main]
 async fn main() -> Result<(), anyhow::Error> {
     let agent_opt = AgentOpt::parse();
-    tedge_utils::logging::initialise_tracing_subscriber(agent_opt.debug);
-
     let tedge_config_location =
         tedge_config::TEdgeConfigLocation::from_custom_root(agent_opt.config_dir.clone());
+
+    let log_level = if agent_opt.debug {
+        tracing::Level::TRACE
```

Comment:
If the ` tedge_utils::logging::set_log_level()` is not changed  setting `TRACE` level here results in `INFO` ;-).

The bug is not here but in ` tedge_utils::logging::set_log_level()` .

Reply:
fixed.